[AMBARI-24551] [Log Search UI] get rid of redundant requests after undoing or redoing several history steps

tobias-istvan opened the pull request:
# What changes were proposed in this pull request?

We can prevent extra calls by using the `LogsContainerService` `resetFiltersForms` method.

## How was this patch tested?

It was tested manually checking the Network tab in dev tools and by running unit tests:
```
PhantomJS 2.1.1 (Mac OS X 0.0.0): Executed 269 of 269 SUCCESS (9.276 secs / 9.178 secs)
✨  Done in 35.58s.
```

Please review [Ambari Contributing Guide](https://cwiki.apache.org/confluence/display/AMBARI/How+to+Contribute) before opening a pull request.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/7/
Test PASSed.

aBabiichuk commented on ambari-logsearch-web/src/app/services/history-manager.service.ts:
```diff
@@ -292,19 +292,12 @@ export class HistoryManagerService {
    */
   private handleUndoOrRedo(value: object): void {
-    const filtersForm = this.logsContainerService.filtersForm;
     this.hasNoPendingUndoOrRedo = false;
```
Looks like `hasNoPendingUndoOrRedo` and `isUndoOrRedo` aren't needed any more

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/9/
Test PASSed.

aBabiichuk commented on ambari-logsearch-web/src/app/services/history-manager.service.ts:
```diff
@@ -209,18 +190,14 @@ export class HistoryManagerService {
     if (startIndex === -1) {
       startIndex = allItems.length;
     }
-    if (endIndex === -1) {
```
This isUndoOrRedo/endIndex logic shouldn't be removed here. It's needed to handle the situation when after some undoing/redoing user goes on with non-historical filters changes, so there's nothing to redo. Sorry for my omission in previous comment.
P.S. Getting rid of hasNoPendingUndoOrRedo looks OK to me.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/21/
Test FAILed.
Test FAILured.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/28/
Test FAILed.
Test FAILured.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/29/
Test FAILed.
Test FAILured.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/38/
Test FAILed.
Test FAILured.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/39/
Test FAILed.
Test FAILured.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/40/
Test FAILed.
Test FAILured.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/41/
Test PASSed.

tobias-istvan commented:
@aBabiichuk
I completely refactored the history manager. The initial reason was to simplify the business logic around the history management, also make it more URL handling compatible since the history is connected to the linkable URL.

I followed the previous approach which is use more action/effects/reducers over services in order to decouple the direct dependencies.

The main reason of the redundant request was the wrong implementation of the `ControlValueAccessor` interface.

I also added some UX improvements to indicate when a request is in progress in the background. There is a global notification and added some specific ones for log tables and graphs.

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/42/
Test PASSed.

aBabiichuk commented on ambari-logsearch-web/src/app/components/filter-history-manager/filter-history-manager.component.ts:
```diff
@@ -0,0 +1,382 @@
+    this.componentsLabels$.takeUntil(this.destroyed$).subscribe(componentsLabels => this.componentsLabelsLocalCopy$.next(componentsLabels));
+    this.activeQueryFieldsLabels$.takeUntil(this.destroyed$).subscribe(fieldLabels => this.activeQueryFieldsLocalCopy$.next(fieldLabels));
+
+    this.activeHistoryItemLabels$.takeUntil(this.destroyed$).subscribe(() => {});
```
What is this for?

aBabiichuk commented on ambari-logsearch-web/src/app/components/action-menu/action-menu.component.ts:
```diff
@@ -74,27 +72,13 @@ export class ActionMenuComponent  implements OnInit, OnDestroy {
-        (clusterName: string) => this.setModalSubmitDisabled(!(!!clusterName))
-      )
+    this.selectedClusterName$.takeUntil(this.destroyed$).subscribe(
+      (clusterName: string) => this.setModalSubmitDisabled(!(!!clusterName))
```
Why not just `setModalSubmitDisabled(!clusterName)`?

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/44/
Test PASSed.
